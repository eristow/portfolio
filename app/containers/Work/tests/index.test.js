/**
 *
 * Tests for Stuff
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { Work } from '../index';

describe('<Work />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<Work dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Work />);
    expect(firstChild).toMatchSnapshot();
  });
});
