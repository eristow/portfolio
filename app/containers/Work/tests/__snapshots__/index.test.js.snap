// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Work /> Should render and match the snapshot 1`] = `
<div>
  <h1>
    My Work
  </h1>
  <div
    class="Work__ButtonContainer-ejvr5h-0 dvjAQO"
  >
    <button
      class="Button-cbu1yy-0 gCZweK"
      color="#03dac6"
      type="button"
    >
      Stock App
    </button>
    <button
      class="Button-cbu1yy-0 hpNGlw"
      color="#03dac6"
      type="button"
    >
      TicTacToe
    </button>
    <button
      class="Button-cbu1yy-0 hpNGlw"
      color="#03dac6"
      type="button"
    >
      Recipe Calculator
    </button>
  </div>
  <h3>
    Web DAW
  </h3>
  <div
    class="Work__ButtonContainer-ejvr5h-0 dvjAQO"
  >
    <button
      class="Button-cbu1yy-0 gCZweK"
      color="#03dac6"
      type="button"
    >
      <a
        href="https://web-daw-frontend.herokuapp.com/"
      >
        Web DAW Project
      </a>
    </button>
    <button
      class="Button-cbu1yy-0 hpNGlw"
      color="#03dac6"
      type="button"
    >
      <a
        href="https://github.com/eristow/web-daw"
      >
        Web DAW Front-End Code
      </a>
    </button>
    <button
      class="Button-cbu1yy-0 hpNGlw"
      color="#03dac6"
      type="button"
    >
      <a
        href="https://github.com/eristow/web-daw-backend"
      >
        Web DAW Back-End Code
      </a>
    </button>
  </div>
</div>
`;
